staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -185,48 +185,51 @@ func DefaultVerifyOptions() x509.VerifyOptions {
 
 // CommonNameUserConversion builds user info from a certificate chain using the subject's CommonName
 var CommonNameUserConversion = UserConversionFunc(func(chain []*x509.Certificate) (*authenticator.Response, bool, error) {
-	if len(chain[0].Subject.CommonName) == 0 {
+	userInfo, succeeded := convertToUserInfo(chain[0].Subject)
+	if !succeeded {
 		return nil, false, nil
 	}
+	return &authenticator.Response{
+		User: userInfo,
+	}, true, nil
+})
 
-	isNewformat := false
-	tenant := ""
-	if len(chain[0].Subject.Organization) > 0 {
-		parts := strings.Split(chain[0].Subject.Organization[0], ":")
-		if len(parts) > 0 && parts[0] == "tenant" {
-			isNewformat = true
-			tenant = parts[1]
-		}
+func convertToUserInfo(subject pkix.Name) (*user.DefaultInfo, bool) {
+	if len(subject.CommonName) == 0 {
+		return nil, false
 	}
-
-	// backward compatible
-	// old format only uses /O
-	//if len(chain[0].Subject.OrganizationalUnit)==0 {
-	if !isNewformat {
-		parts := strings.Split(chain[0].Subject.CommonName, ":")
-		if len(parts) > 1 {
-			tenant = parts[0]
-		}
-		return &authenticator.Response{
-			User: &user.DefaultInfo{
-				Name:   chain[0].Subject.CommonName,
-				Groups: chain[0].Subject.Organization,
+	tenant := ""
+	if len(subject.Organization) > 0 {
+		parts := strings.Split(subject.Organization[0], ":")
+		if parts[0] == "tenant" {
+			// new format (tenant string from Organization)
+			if len(subject.Organization) > 1 {
+				// with new format, tenant comes from
+				// Organization and only one is allowed
+				return nil, false
+			}
+			numOfParts := len(parts)
+			if numOfParts == 2 {
```

Comment:
The cases that the numOfParts is 1 is not handled. So if subject.Organization[0]=="tenant", we will get a userInfo where the tenant is ""

Reply:
that's the current defined behavior. see test case "new format, no colon tenant in organization"